Directly link to a help tab.

pfefferle opened the pull request:
This PR allows you to directly link to help tabs. The idea is to use it for the Onboarding as part of a launchpad.

<!--- Provide a general summary of your changes in the Title above -->

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
*

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Open the ActivityPub admin page with the hash of the help tab you want to open. The link for the "Account Migration" help tab is `/wp-admin/options-general.php?page=activitypub#tab-link-account-migration` for example.

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [X] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [X] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Enable direct linking to Help Tabs.

</details>

pfefferle committed:
Directly link to a help tab.

This PR allows you to directly link to help tabs. The idea is to use it for the Onboarding as part of a launchpad.

pfefferle committed:
phpcs fixes

matticbot committed:
Add changelog

pfefferle committed:
Merge branch 'trunk' into add/link-to-helptab

obenland commented on includes/wp-admin/class-admin.php:
```diff
@@ -60,6 +60,8 @@ public static function init() {
 			\add_action( 'tool_box', array( self::class, 'tool_box' ) );
 		}
+
+		\add_action( 'admin_print_footer_scripts', array( self::class, 'open_help_tab' ) );
```
Would it still work if we used `"admin_print_footer_scripts-{$hook_suffix}"` and limited it to our settings page?
If we can limit it to the settings page, maybe the callback should live in class-settings?

obenland commented on includes/wp-admin/class-admin.php:
```diff
@@ -609,4 +611,28 @@ public static function plugin_update_message( $data, $update ) {
+					$(hash + ' a').click();
+				}, 500);
+			}
+		});
```
I got a jQery migrate warning in my testing, `jQuery.fn.click() event shorthand is deprecated`.
Could we just use vanilla JS for this snippet?

obenland reviewed (commented):
Nice! Small, yet powerful.

I tested it with the path you suggested in the issue description but it didn't work immediately, I had to change the hash to `#tab-link-account-migration`. 

That made me think that we might want to add an allow list of links that can automatically be clicked, so users can't be tricked into clicking unintended links by getting passed malicious hashes?

pfefferle committed:
use `{$hook_suffix}` action

pfefferle committed:
use vanilla js

pfefferle committed:
add allowlist (limited to activitypub help tabs)

pfefferle committed:
fix phpcs issues

obenland committed:
Backslashit

obenland committed:
Increase selector specificity

Only clicks anchors that are direct descendants of `hash`, where the href attribute begins with `#tab-panel-`.
